packages/service/ni_measurement_plugin_sdk_service/_internal/grpc_servicer.py

```diff
@@ -271,10 +275,21 @@ def _serialize_response(
         outputs: Any,
     ) -> v1_measurement_service_pb2.MeasureResponse:
         return v1_measurement_service_pb2.MeasureResponse(
-            outputs=_serialize_outputs(
-                self._output_metadata, outputs, self._service_info.service_class + ".Outputs"
-            )
+            outputs=_serialize_outputs(self._output_metadata, outputs, self._outputs_message_type)
+        )
+
+    def _validate_parameters(self, request: v1_measurement_service_pb2.MeasureRequest) -> None:
+        pool = descriptor_pool.Default()
+        configuration_proto = pool.FindMessageTypeByName(
+            self._configuration_parameters_message_type
         )
+        configuration_message = message_factory.GetMessageClass(configuration_proto)()
+        if not request.configuration_parameters.Is(configuration_message.DESCRIPTOR):
+            expected = "type.googleapis.com/" + configuration_message.DESCRIPTOR.full_name
+            actual = request.configuration_parameters.type_url
+            sys.stderr.write(f"Note: Wrong message type. Expected {expected!r} but got {actual!r}")
+        elif not request.configuration_parameters.Unpack(configuration_message):
```

Comment:
Is this unpacking the message and throwing it away, for validation purposes? I don't think that's necessary. All `Unpack` does is use `Is` to validate the type URL then `ParseFromString`, which I assume we are still doing elsewhere since this is a validate function and it doesn't return the unpacked message.

https://github.com/protocolbuffers/protobuf/blob/9f08ec543be4a38f7831b5b7cea07fe9015cb3c2/python/google/protobuf/internal/well_known_types.py#L60

Reply:
Yeah, I was just doing it for validation purposes, thinking it would validate the parameters. But since it does nothing more than `Is` and `ParseFromString`, I'll drop it.